common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java

```diff
@@ -102,7 +103,17 @@ public R getEntity(String idField, String value) {
     public QueryResult<R> search(SearchRequest request) {
         SolrRequest solrRequest = createSearchSolrRequest(request);
         QueryResult<D> results = repository.searchPage(solrRequest, request.getCursor());
-        Stream<R> converted = results.getContent().map(entryConverter).filter(Objects::nonNull);
+        Stream<R> converted;
```

Comment:
We could extract line 105 that was in the old code into a new method, and override it in StoreStreamerSearchService. And you could also move mapToThinEntry to StoreStreamerSearchService class and make it an abstract. This way every Service that uses Voldemort (Store) must implement it.